packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -423,6 +427,172 @@ class ActionDispatcher with Diagnosticable {
 /// Actions are typically invoked using [Actions.invoke] with the context
 /// containing the ambient [Actions] widget.
 ///
+/// {@tool dartpad --template=stateful_widget_scaffold_center}
+///
+/// This example creates a custom [Action] subclass `ModifyAction` for modifying
+/// a model, and another, `SaveAction` for saving it.
+///
+/// One thing that this example demonstrates is passing arguments to the
+/// [Intent] to be carried to the [Action]. This shows how actions can get data
+/// either from their own construction (like the `model` in this example), or
+/// from the intent passed to them when invoked (like the increment `amount` in
+/// this example).
+///
+/// This example also shows how to insulate a widget from knowing too much about
```

Comment:
This example also shows how to insulate a widget from knowing too much about its surroundings using Intents.

This example also demonstrates how to use Intents to limit a widget's dependencies on its surroundings.